linera-core/src/chain_worker/actor.rs

```diff
@@ -165,39 +165,39 @@ where
 {
     /// Spawns a new task to run the [`ChainWorkerActor`], returning an endpoint for sending
     /// requests to the worker.
-    pub async fn spawn(
+    pub fn spawn(
         config: ChainWorkerConfig,
         storage: StorageClient,
         certificate_value_cache: Arc<ValueCache<CryptoHash, HashedCertificateValue>>,
         blob_cache: Arc<ValueCache<BlobId, HashedBlob>>,
         chain_id: ChainId,
-        join_set: &mut JoinSet<()>,
-    ) -> Result<mpsc::UnboundedSender<ChainWorkerRequest<StorageClient::Context>>, WorkerError>
-    {
+        receiver: mpsc::UnboundedReceiver<ChainWorkerRequest<StorageClient::Context>>,
+        join_set: &mut impl JoinSetExt,
+    ) {
         let (service_runtime_thread, execution_state_receiver, runtime_request_sender) =
             Self::spawn_service_runtime_actor(chain_id);
 
-        let worker = ChainWorkerState::load(
-            config,
-            storage,
-            certificate_value_cache,
-            blob_cache,
-            chain_id,
-            execution_state_receiver,
-            runtime_request_sender,
-        )
-        .await?;
-
-        let (sender, receiver) = mpsc::unbounded_channel();
-        let actor = ChainWorkerActor {
-            worker,
-            incoming_requests: receiver,
-            service_runtime_thread,
-        };
+        join_set.spawn_task(Box::pin(async move {
+            let state = ChainWorkerState::load(
+                config,
+                storage,
+                certificate_value_cache,
+                blob_cache,
+                chain_id,
+                execution_state_receiver,
+                runtime_request_sender,
+            )
+            .await
+            .expect("chain worker actor shouldn't be spawned for non-existent chain");
```

Comment:
I'm not sure about this. We should probably consider carefully what would happen on a storage error. An alternative might be to change the `WorkerState` to do the spawning instead. The only change in this file would be to change `spawn` to be `new`, with the changes to the signature that you already did, plus removing the `join_set` parameter.

Reply:
But with the suggestion below we don't need to `expect` the error :thinking:

Reply:
Oh, sorry, I think I missed the crux of your suggestion.  You're right!  Let's do that :)